### PR TITLE
Add nightly playground

### LINKED ISSUE #129: Create a "nightly" instance of Playground that gets updated everyday with the daily build.
### Is your feature request related to a problem? Please describe

I'd like to be able to play with new features faster as they're getting build on OpenSearch.  Therefore, I'd like a "nightly" version of playground to get built with every successful build -- first with the 2.x build, but ultimately with main as well. 

### Describe the solution you'd like

Whenever there's a successful night built, we replace the current nightly playground build automatically.   We will need a way to see what version we're pointing at. 

### Acceptance Criteria:

1. Support upcoming version for 2.x
2. OS and OSD gets deployed nightly on a regular basis using the latest builds.
3. Have generic anonymous(read-only) access for OpenSearch DashBoards.
4. Display the commits related to a build-id.
5. The instance should be publicly accessible.

### Action Items
- [x] https://github.com/opensearch-project/opensearch-devops/issues/130
- [x] https://github.com/opensearch-project/opensearch-cluster-cdk/issues/72
- [x] https://github.com/opensearch-project/opensearch-cluster-cdk/issues/73
- [x] https://github.com/opensearch-project/opensearch-cluster-cdk/issues/74
- [x] https://github.com/opensearch-project/opensearch-cluster-cdk/issues/75 : Research and integrate opensearch-cluster-cdk into nightly playgrounds. May involve publishing the cdk code base to npm or consume it by cloning the repo. 
- [x] https://github.com/opensearch-project/opensearch-devops/issues/131
- [x] https://github.com/opensearch-project/opensearch-devops/issues/132
- [x] Set up accounts and required permissions (OIDC) to deploy the stacks on the infrastructure
- [x] Ensure the deploying artifact validation always validates with security plugin strictly
- [x] Add GHA deployment workflow for tools and notification stack
- [x] Add GHA deployment workflow for opensearch-cluster-cdk stacks to deploy nightly.

### PR DESCRIPTION
**Pull request by gaiksaya**
### Description
This change adds the initial draft of nightly playgrounds with below specifics:
1. distribution version, distribution urls are grabbed on the go whereas all other important parameters are hardcoded in the code
2. Adds tests
3. Adds GHA workflow to run tests

### Issues Resolved
#129 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: .github/workflows/nightly-playground-test.yml [hunk 1]
```diff
@@ -0,0 +1,35 @@
+name: Nightly Playground Build and Test
+
+on:
+  push:
+    paths:
+      - nightly-playground/**
```
Comment (peterzhuamazon) [inline .github/workflows/nightly-playground-test.yml:6]: Think pr trigger is enough here.

File: nightly-playground/lib/nightly-playground-stack.ts [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+export class NightlyPlaygroundStack {
+  public stacks: Stack[] = []; // only required for testing purpose
+
+  constructor(scope: Construct, id: string, props: StackProps) {
+    const distVersion = scope.node.tryGetContext('distVersion');
+    if (distVersion === 'undefined') {
+      throw new Error('distVersion parameter cannot be empty! Please provide the OpenSearch distribution version');
+    }
+    const distributionUrl = scope.node.tryGetContext('distributionUrl');
```
Comment (peterzhuamazon) [inline nightly-playground/lib/nightly-playground-stack.ts:21]: Can we make the variable naming convention consistent?

File: nightly-playground/lib/nightly-playground-stack.ts [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    if (distributionUrl === 'undefined') {
+      throw new Error('distributionUrl parameter cannot be empty! Please provide the OpenSearch distribution URL');
+    }
+    const dashboardsUrl = scope.node.tryGetContext('dashboardsUrl');
```
Comment (peterzhuamazon) [inline nightly-playground/lib/nightly-playground-stack.ts:25]: Probably call the distVersion or distributionUrl as opensearchVersion or opensearchUrl, to be consistent with dashboardsUrl?

**Review by peterzhuamazon — COMMENTED** [review]
So from the looks of it, this is basically import cluster cdk, and deploy a stack of OpenSearch / Dashboards with provided url?

Just wondering why cant we use the existing cluster cdk directly?

Thanks.

File: nightly-playground/lib/nightly-playground-stack.ts (hunk 3 above)
Comment (gaiksaya) [reply nightly-playground/lib/nightly-playground-stack.ts:25]: Its is consistent with what we have for opensearch-cluster-cdk

File: nightly-playground/lib/nightly-playground-stack.ts (hunk 2 above)
Comment (gaiksaya) [reply nightly-playground/lib/nightly-playground-stack.ts:21]: Same as below

**Comment by gaiksaya** [comment]
> So from the looks of it, this is basically import cluster cdk, and deploy a stack of OpenSearch / Dashboards with provided url?
> 
> Just wondering why cant we use the existing cluster cdk directly?
> 
> Thanks.

The reason we went with this approach is to reduce the context variables and source code what is required for this use case. Such as security should always be enabled. Also, for future stacks, we can extend / depend on this base stacks. Another reason was to stick to 1 version of opensearch-cluster-cdk. The project is pretty new and has active changes coming in more often which we need not be bothered unless required.

**Review by peterzhuamazon — APPROVED** [review]
@gaiksaya confirms that mend error is not related.
Thanks, approved.

**Pull request merged**